src/server/worker/datum_iterator.go

```diff
@@ -40,9 +40,13 @@ func newPFSDatumIterator(pachClient *client.APIClient, input *pps.PFSInput) (Dat
 		// before all commits have inputs
 		return result, nil
 	}
+	impliedGlob := input.Glob
+	if input.S3 {
+		impliedGlob = "/"
+	}
 	fs, err := pachClient.GlobFileStream(pachClient.Ctx(), &pfs.GlobFileRequest{
 		Commit:  client.NewCommit(input.Repo, input.Commit),
-		Pattern: input.Glob,
+		Pattern: impliedGlob,
```

Comment:
Instead of this glob fixing logic, should we just fail for any glob that's not `/` on s3 inputs? That may help users understand what's getting mounted in s3g.

Reply:
That makes sense! Done.